Wikipedia/assets/codemirror/codemirror-index.html

```diff
@@ -611,7 +614,45 @@
       editor.replaceSelection(`${char} ${editor.getSelection().replace(/\n/gi, `\n${char} `)}`, 'around')
     }
 
+    const verticalOffsetNeededToScrollCursorIntoView = () => {
+      const fromTopOfWebView = cursorOffsetFromTopOfWebView()
+      if (fromTopOfWebView < 0) {
+        return fromTopOfWebView
+      }
+      const fromTopOfKeyboard = cursorOffsetFromTopOfKeyboard()
+      if (fromTopOfKeyboard > 0) {
+        return fromTopOfKeyboard
+      }
+      return 0
+    }
+
+    let keyboardHeight = 0
+    const webViewHeightNotCoveredByKeyboard = () => window.innerHeight - keyboardHeight
+    const cursorOffsetFromTopOfWebView = () => editor.cursorCoords().top - window.pageYOffset
+    const cursorOffsetFromTopOfKeyboard = () => editor.cursorCoords().bottom - window.pageYOffset - webViewHeightNotCoveredByKeyboard()
+
+    const scrollToCenterCursorVerticallyIfNeeded = () => {
+      const offset = verticalOffsetNeededToScrollCursorIntoView()
+      if (offset === 0) {
+        return
+      }
+      window.scrollBy(0, offset)
+    }
+
+    let resizeTimer
+    window.addEventListener('resize', () => {
+      clearTimeout(resizeTimer)
+      resizeTimer = setTimeout(() => {
+        scrollToCenterCursorVerticallyIfNeeded()        
+      }, 250)
+    }, true)
+
+
     wmf.commands = {
+      keyboardHeightChanged: (newHeight) => {
```

Comment:
You could monitor changes to `adjustedContentInset` with the `UIScrollViewDelegate` method `scrollViewDidChangeAdjustedContentInset` instead of the keyboard notification and it'd take into account any overlays that aren't the keyboard (for example if the edit toolbar moves into the view itself to keep it visible even after the keyboard is dismissed). You could also pass the full insets to JS to account for any overlays we'd add to the top.

Reply:
I like it!